### DISCUSSION TITLE
Can custom Go code be integrated as plugins?

### DISCUSSION THREAD

**Discussion by sheldonhull** [go-task/task#1761] [Q&A] [answered]
I think the answer is no, but it's been years since I last researched this with task, so I'm checking if anything changed that would allow easily including custom Go files as tasks.
Primary use case would be where the basics in task don't work to create a `<foo>.go` in a directory and then be able to use this as a remote task library that included custom plugin code.

I use mage right now, so this is essentially saying could I plug in some custom functions and call via task natively, or not.

I did dig back through discussions and issues, and couldn't find anything specific, only recalling that as of a few years ago it wasn't supported.

**Comment by andreynering** [comment, marked as answer]
This is not supported. You can call a Go script manually via `go run script.go` as a workaround.

**Reply by sheldonhull** [reply]
thanks for the quick update. Keep up the good work 🥇
